aws-traffic-mirroring/templates/vpc-mirroring-target-instance.yaml

```diff
@@ -0,0 +1,151 @@
+    Default: 600
+  MaxFileSize:
+    Type: String
+    Description: Max file size before creating a new pcap file
+    Default: 50
+
+Resources:
+  SecurityGroup:
+    Type: "AWS::EC2::SecurityGroup"
+    Properties:
+      GroupDescription: !Sub Security group for ${AWS::StackName} instance
+      GroupName: !Sub ${AWS::StackName}-sg
+      VpcId: !Ref CustomerVpcId
+      SecurityGroupIngress:
+        - { IpProtocol: udp, FromPort: 4789, ToPort: 4789, CidrIp: !Ref SourceVpcIpv4Cidr, Description: 'Allow mirrored traffic from VPC' }
+      Tags:
+        - { Key: Name, Value: !Sub '${AWS::StackName}-sg' }
+
+  Role:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          -
+            Effect: Allow
+            Principal:
+              Service: ec2.amazonaws.com
+            Action: sts:AssumeRole
+      Path: /
+      RoleName: !Sub ${AWS::StackName}-role
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+
+  RoleProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      Path: /
+      Roles:
+        - !Ref Role
+      InstanceProfileName: !Sub '${AWS::StackName}-profile'
+
+  LaunchTemplate:
+    Type: "AWS::EC2::LaunchTemplate"
+    Metadata:
+      AWS::CloudFormation::Init:
+        config:
+          files:
+            /home/seekret-conf.env:
+              content: !Sub |
+                  ACCESS_KEY=${BucketAccessKey}
+                  SECRET_ACCESS_KEY=${BucketSecretKey}
+                  S3_URL=${BucketS3Url}
+                  BUCKET_NAME=${BucketName}
+                  FOLDER_NAME=${BucketFolderName}
+                  BPF_FILTER=tcp and port 4789
+                  MAX_FILE_SIZE=${MaxFileSize}
+                  ROTATION_SECONDS=${RotationSeconds}
+              mode: "000644"
+              owner: "ec2-user"
+              group: "ec2-user"
+    Properties:
+      LaunchTemplateName: !Sub ${AWS::StackName}-lt
+      LaunchTemplateData:
+        EbsOptimized: true
+        IamInstanceProfile:
+          Arn: !GetAtt RoleProfile.Arn
+        ImageId: !Ref LatestAmiId
+        InstanceMarketOptions:
+          MarketType: spot
+          SpotOptions:
+            SpotInstanceType: one-time
+        InstanceType: m5.large
+        SecurityGroupIds:
+          - !Ref SecurityGroup
+        TagSpecifications:
+          - ResourceType: instance
+            Tags:
+              - { Key: Name, Value: !Sub '${AWS::StackName}' }
+          - ResourceType: volume
+            Tags:
+              - { Key: Name, Value: !Sub '${AWS::StackName}-volume' }
+        UserData:
+          Fn::Base64: !Sub |
+            #!/bin/bash
+            set -o xtrace
+            sudo yum update -y
+
+            sudo /opt/aws/bin/cfn-init -v --stack ${AWS::StackName} --resource LaunchTemplate --region ${AWS::Region}
+
+            # Setup SSM for remote management
+            yum install -y amazon-ssm-agent
```

Comment:
sudo

Reply:
actually it is not required.. maybe so the regular user could connect via AWS ssm?

Reply:
added anyway